opensquirrel/exceptions.py

```diff
@@ -0,0 +1,20 @@
+"""This module contains all custom exception used by ``OpenSquirrel``."""
+
+from typing import Any
+
+
+class UnsupportedGateError(Exception):
+    """Should be raised when a gate is not supported."""
+
+    def __init__(self, gate: Any, *args: Any) -> None:
+        """Init of the ``UnsupportedGateError``.
+
+        Args:
+            gate: Gate that is not supported.
+        """
+        self.unsupported_gate = gate
```

Comment:
I wouldn't keep `unsupported_gate` as a member, but just the message that is passed  to `super().__init__`.